scripts/cat/pelts.py

```diff
@@ -1,626 +1,1072 @@
+                 pattern:str=None,
+                 tortiepattern:str=None,
+                 vitiligo:str=None,
+                 points:str=None,
+                 accessory:str=None,
+                 paralyzed:bool=False,
+                 opacity:int=100,
+                 scars:list=None,
+                 tint:str="none",
+                 skin:str="BLACK",
+                 white_patches_tint:str="none",
+                 kitten_sprite:int=None,
+                 adol_sprite:int=None,
+                 adult_sprite:int=None,
+                 senior_sprite:int=None,
+                 para_adult_sprite:int=None,
+                 reverse:bool=False,
+                 ) -> None:
+        self.name = name
         self.colour = colour
-        self.length = length
-        self.white = True
-
-    def __repr__(self):
-        return f"white and {self.colour}{self.length}"
-
-
-class Tabby():
-    name = "Tabby"
-    sprites = {1: 'tabby', 2: 'white'}
+        self.white_patches = white_patches
+        self.eye_colour = eye_color
+        self.eye_colour2 = eye_colour2
+        self.tortiebase = tortiebase
+        self.pattern = pattern
+        self.tortiepattern = tortiepattern
+        self.tortiecolour = tortiecolour
+        self.vitiligo = vitiligo
+        self.length=length
+        self.points = points
+        self.accessory = accessory
+        self.paralyzed = paralyzed
+        self.opacity = opacity
+        self.scars = scars if isinstance(scars, list) else []
+        self.tint = tint
+        self.white_patches_tint = white_patches_tint
+        self.cat_sprites =  {
+            "kitten": kitten_sprite if kitten_sprite is not None else 0,
+            "adolescent": adol_sprite if adol_sprite is not None else 0,
+            "young adult": adult_sprite if adult_sprite is not None else 0,
+            "adult": adult_sprite if adult_sprite is not None else 0,
+            "senior adult": adult_sprite if adult_sprite is not None else 0,
+            "senior": senior_sprite if senior_sprite is not None else 0,
+            "para_adult": para_adult_sprite if para_adult_sprite is not None else 0,
+        }        
+        self.cat_sprites['newborn'] = 20
+        self.cat_sprites['para_young'] = 17
+        self.cat_sprites["sick_adult"] = 18
+        self.cat_sprites["sick_young"] = 19
+        
+        self.reverse = reverse
+        self.skin = skin
 
-    def __init__(self, colour, white, length):
-        self.white = white  # boolean; does cat have white on it or no
-        self.colour = colour
-        self.length = length
+    @staticmethod
+    def generate_new_pelt(gender:str="female", parents:tuple=(), age:str="adult"):
+        new_pelt = Pelt()
+        
+        pelt_white = new_pelt.init_pattern_color(parents, gender)
+        new_pelt.init_white_patches(pelt_white, parents)
+        new_pelt.init_sprite()
+        new_pelt.init_scars(age)
+        #self.init_accessories(age)
```

Comment:
It is commented out because the original function actually didn't work, and I was worried that making it work would break something down the line.  It self a variable called "acc_display" , which is never referenced anywhere else in the code - not even in the __init__ for the cat. I'm guessing it allowed that new parameter to be set because this function was being called in __init__.

Reply:
If it is not used, we should clean it up and remove the function + the comment line with the function.
In my opinion, there is no need to store old unused code, these are just unnecessary lines of code.

Reply:
I did some testing, and I think it's safe to comment out the line and allow accessory generation.  Since no accessories generating was a bug, it's probably best to fix it. I did lower the chances of cats generating with accessories, since it was set pretty high.